x-pack/legacy/plugins/code/public/components/commits/commit.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import {
+  EuiButtonIcon,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiPanel,
+  EuiText,
+  EuiTextColor,
+  EuiCopy,
+  EuiTitle,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { CommitInfo } from '../../../model/commit';
+import { MAIN_ROOT } from '../routes';
+import { PathTypes } from '../../common/types';
+
+const COMMIT_ID_LENGTH = 8;
+
+interface ActionProps {
+  repoUri: string;
+  commitId: string;
+}
+
+const copyLabel = i18n.translate('xpack.code.commits.copyCommitSha', {
+  defaultMessage: 'Copy the full commit ID',
+});
+const revisionLinkLabel = i18n.translate('xpack.code.commits.revisionLink', {
+  defaultMessage: 'View the project at this commit',
+});
+
+const CommitActions = ({ commitId, repoUri }: ActionProps) => {
+  const revisionUri =
+    '#' +
+    MAIN_ROOT.replace(/:pathType\(.*\)/, PathTypes.tree)
+      .replace(':resource/:org/:repo', repoUri)
+      .replace(':revision', commitId);
+
+  return (
+    <div className="commit__actions">
+      <EuiCopy textToCopy={commitId}>
+        {copy => <EuiButtonIcon aria-label={copyLabel} onClick={copy} iconType="copyClipboard" />}
+      </EuiCopy>
+      <EuiButtonEmpty
+        className="commit__diff-link"
+        href={revisionUri}
+        aria-label={revisionLinkLabel}
+      >
+        <EuiText size="s">{commitId}</EuiText>
+      </EuiButtonEmpty>
+      <EuiButtonIcon href={revisionUri} aria-label={revisionLinkLabel} iconType="editorCodeBlock" />
+    </div>
+  );
+};
+
+interface Props {
+  commit: CommitInfo;
+  date: string;
+  repoUri: string;
+}
+
+export const Commit = (props: Props) => {
+  const { date, commit, repoUri } = props;
+  const { message, committer, id } = commit;
+  const [title, ...rawDescription] = message.split('\n');
+  const description = rawDescription.join('\n').trim();
```

Comment:
This is based on standard git commit formatting. I'm unsure if there is/should be a helper for something like this.

Reply:
yes, having a util function to handle this would be great.

Reply:
ef18c07